### PR TITLE
all hail shield

### PR DESCRIPTION
**Pull request by john-b-edwards**
resolving #48, #50, #51, #52

height formatting:
```r
nflreadr::load_rosters(T) |>
    dplyr::pull(height) |>
    table()
#> 
#>    61    63    64    65    66    67    68    69    70    71    72    73    74 
#>     1     2    17    42   175   464  1596  3505  7823 10789 14645 16405 18773 
#>    75    76    77    78    79    80    81    82    84 
#> 17871 14236 10090  4705  1838   599   146    11     1
```

missing birthdates (still missing some but the number is far more reasonable):
```r
nflreadr::load_rosters(T) |>
    dplyr::pull(birth_date) |>
    is.na() |>
    sum()
#> [1] 2146
```

full names display properly:
```r
nflreadr::load_rosters(1999:2022) |>
    dplyr::filter(gsis_id == "00-0029263") |>
    dplyr::select(season, full_name)
#> ── nflverse roster data ────────────────────────────────────────────────────────
#> ℹ Data updated: 2022-08-18 13:10:12 PDT
#> # A tibble: 11 × 2
#>    season full_name     
#>     <int> <chr>         
#>  1   2012 Russell Wilson
#>  2   2013 Russell Wilson
#>  3   2014 Russell Wilson
#>  4   2015 Russell Wilson
#>  5   2016 Russell Wilson
#>  6   2017 Russell Wilson
#>  7   2018 Russell Wilson
#>  8   2019 Russell Wilson
#>  9   2020 Russell Wilson
#> 10   2021 Russell Wilson
#> 11   2022 Russell Wilson
```

headshot urls fixed

```r
nflreadr::load_rosters(1999:2022) |>
    dplyr::filter(gsis_id == "00-0029263") |>
    dplyr::select(season, headshot_url)
#> ── nflverse roster data ────────────────────────────────────────────────────────
#> ℹ Data updated: 2022-08-18 13:10:12 PDT
#> # A tibble: 11 × 2
#>    season headshot_url                                                          
#>     <int> <chr>                                                                 
#>  1   2012 https://static.www.nfl.com/image/private/f_auto,q_auto/league/lszslgb…
#>  2   2013 https://static.www.nfl.com/image/private/f_auto,q_auto/league/lszslgb…
#>  3   2014 https://static.www.nfl.com/image/private/f_auto,q_auto/league/lszslgb…
#>  4   2015 https://static.www.nfl.com/image/private/f_auto,q_auto/league/lszslgb…
#>  5   2016 https://static.www.nfl.com/image/private/f_auto,q_auto/league/rrpzsqy…
#>  6   2017 https://static.www.nfl.com/image/private/f_auto,q_auto/league/rrpzsqy…
#>  7   2018 https://static.www.nfl.com/image/private/f_auto,q_auto/league/rrpzsqy…
#>  8   2019 https://static.www.nfl.com/image/private/f_auto,q_auto/league/rrpzsqy…
#>  9   2020 https://static.www.nfl.com/image/private/f_auto,q_auto/league/r9z2h3e…
#> 10   2021 https://static.www.nfl.com/image/private/f_auto,q_auto/league/lszslgb…
#> 11   2022 https://static.www.nfl.com/image/private/f_auto,q_auto/league/lszslgb…
```

### CODE REVIEW

File: src/update_roster.R [hunk 1]
```diff
@@ -259,16 +260,16 @@ build_rosters_weekly_ngsapi <- function(season){
           game_type %in% c("WC", "DIV", "CON", "SB") ~ week + max(week[game_type == "REG"]),
           T ~ week
         ),
-
       )
-    weekly_rosters[["birth_date"]] <- NULL # sometimes ngsscrapR::scrape_rosters() returns this, sometimes it doesn't
+    weekly_rosters[["birth_date"]] <-
```
Comment (tanho63) [inline src/update_roster.R:264]: Why blanking birthdates here? that makes 2016+ rosters not have birth dates?

Comment (john-b-edwards) [reply src/update_roster.R:264]: For some reason `ngsscrapR` will sometimes return a `birth_date` column and sometimes will not. I just delete this column and pull `birth_date` from shield:

```r
nflreadr::load_rosters(2016) |>
    dplyr::pull(birth_date) |>
    head()
#> [1] "1985-03-22" "1986-12-11" "1990-07-11" "1990-06-17" "1991-09-01"
#> [6] "1992-05-13"
```

Comment (tanho63) [reply src/update_roster.R:264]: hmm. but you don't run shield on players entering league after 2016 then?

Comment (john-b-edwards) [reply src/update_roster.R:264]: Sorry, I don't follow -- I build shield rosters for pre-2002, build shield and data exchange rosters for 2002-2015, and build shield and ngs rosters for 2016+. Then I join shield to the data exchange and ngs rosters to get cleaned values for height, weight, birthdate, etc.

File: src/update_roster.R [hunk 2]
```diff
@@ -281,8 +282,7 @@ convert_weekly_to_season_rosters <- function(weekly_rosters){
   return(roster)
 }
 
-fill_ids <- function(roster){
-
+fill_ids <- function(roster) {
```
Comment (tanho63) [inline src/update_roster.R:285]: Something I wanted to do here but didn't get to was running

```r
df %>%
  group_by(gsis_id) %>% 
  fill(contains("id"), birth_date, .direction = "downup") %>% 
  ungroup()
```

File: src/update_roster.R (hunk 1 above)
Comment (tanho63) [reply src/update_roster.R:264]: Oh! I did not read L363. Duh.

**Review by tanho63 — APPROVED** [review]

**Pull request merged**